add filter config document (#1755)
vdaas/vald: docs/user-guides/filtering-configuration.md

# Filter Configuration

This page describes how enabled filtering features for your Vald cluster.

## Requirement

To use any filtering functions with the Vald cluster, you must deploy the ingress or egress filter component before deploying the Vald cluster.

The filter component can be deployed anywhere, but it must be able to communicate with the Vald Filter gateway.

> Every filter component which you'd like to use as the Vald filter component must apply Vald's filter gRPC interface.

Author: ```suggestion
Every filter component must meet  Vald's filter gRPC interface.
```

> Every filter component which you'd like to use as the Vald filter component must apply Vald's filter gRPC interface.

Reviewer: ```suggestion
Every filter component should meet Vald's filter gRPC interface.
```

- The egress rpc definition

    ```rpc
    service Filter {
      // Represent the RPC to filter the distance.
      rpc FilterDistance(payload.v1.Object.Distance)
          returns (payload.v1.Object.Distance) {
        option (google.api.http) = {
          post : "/filter/egress/distance"
          body : "*"
        };
      }
      // Represent the RPC to filter the vector.
      rpc FilterVector(payload.v1.Object.Vector)
          returns (payload.v1.Object.Vector) {
        option (google.api.http) = {
          post : "/filter/egress/vector"
          body : "*"
        };
      }
    }
    ```

For more details, please refer to [the Vald Filter Gateway document](../overview/component/filter-gatewya.md)

### Official filter components

Vald provides the two types of ingress filter components.
These will help you to implement your original filter component.

Please refer to:
- [Vald ONNX Ingress Filter](https://github.com/vdaas/vald-onnx-ingress-filter)
- [Vald Tensorflow Ingress Filter](https://github.com/vdaas/vald-tensorflow-ingress-filter)

## Configuration

It is easy to enable the filtering feature.

```yaml
...
gateway:
...
    filter:
        enabled: true
...
```

> It is because the Vald Filter gateway connects the filter component specified in the users' request.

Reviewer: ```suggestion
The Vald Filter gateway connects to the filter component specified in the users' request.
So, you can use the filtering function only by setting `gateway.filter.enabled=true` in your Helm chart.```

    ```rpc
    service Filter {
      // Represent the RPC to generate the vector.
      rpc GenVector(payload.v1.Object.Blob) returns (payload.v1.Object.Vector) {
        option (google.api.http) = {
          post : "/filter/ingress/object"
          body : "*"
        };
      }
      // Represent the RPC to filter the vector.
      rpc FilterVector(payload.v1.Object.Vector)
          returns (payload.v1.Object.Vector) {
        option (google.api.http) = {
          post : "/filter/ingress/vector"
          body : "*"
        };
      }
    }
    ```

- The egress RPC definition

    ```rpc
    service Filter {
      // Represent the RPC to filter the distance.
      rpc FilterDistance(payload.v1.Object.Distance)
          returns (payload.v1.Object.Distance) {
        option (google.api.http) = {
          post : "/filter/egress/distance"
          body : "*"
        };
      }
      // Represent the RPC to filter the vector.
      rpc FilterVector(payload.v1.Object.Vector)
          returns (payload.v1.Object.Vector) {
        option (google.api.http) = {
          post : "/filter/egress/vector"
          body : "*"
        };
      }
    }
    ```

> 

Reviewer 2: should we describe more about difference between Ingress Filter and Egress Filter
 and it's usecase.

Author: Could you please tell me more details?
Isn't it enough to show the filter gateway component?
